pyproject.toml

```diff
@@ -66,7 +66,7 @@ joblib = "*"
 plotly = "*"
 hydra_slayer = "*"
 matplotlib = "*"
-typing_extensions = "*"
+typing_extensions = ">=4.1.0,<4.6.0"
```

Comment:
Why are we limiting upper version with 4.6.0? The last version is 4.12.2, does it cause any problems?

Reply:
It should work with 4.12.2 version as well. The upper bound is set so lock file stays the same.

Reply:
If it works on `4.12.2` then we should allow it to have this version by setting:
```
typing_extensions = "^4.1.0"
```

What is the problem with updating lock?